libraries/chain/block_header_state.cpp

```diff
@@ -203,20 +205,35 @@ block_header_state block_header_state::next(const signed_block_header& h, valida
       .new_protocol_feature_activations = std::move(new_protocol_feature_activations)
    };
 
+   digest_type action_mroot; // digest_type default value is empty
+
+   if (h.is_proper_svnn_block()) {
+      // if there is no Finality Tree Root associated with the block,
+      // then this needs to validate that h.action_mroot is the empty digest
+      auto next_core_metadata = core.next_metadata(if_ext.qc_claim);
+      auto no_finality_tree_associated = core.is_genesis_block_num(next_core_metadata.final_on_strong_qc_block_num);
```

Comment:
Could also just use `bool` instead of `auto` to make it more clear to the reader.